testsuite/tests/letrec-check/unboxed.ml

```diff
@@ -20,10 +20,10 @@ Line 2, characters 12-19:
 Error: This kind of expression is not allowed as right-hand side of `let rec'
 |}];;
 
-type r = A of r [@@unboxed]
+type r = A of r [@@unboxed] [@@value]
```

Comment:
It does not.  It's a tricky case, not handled by the existing defaulting for type variables (none here) or abstract types (it has a kind).  Since circular types like this are useless anyway and unlikely to occur in practice, and handling it well would require some careful thought, it seemed best to just require an annotation.  I wish the type checker would just reject circular unboxed types, frankly.